src/platform_impl/windows/window.rs

```diff
@@ -356,6 +356,21 @@ impl Window {
         Ok(())
     }
 
+    #[inline]
+    pub fn set_drag_window(&self) -> Result<(), ExternalError> {
+        unsafe {
+            winuser::ReleaseCapture();
```

Comment:
Should this set `window_state.mouse.capture_count` to `0`?

Reply:
You should test for regressions for #1830, #1797, and #1796.

You should also check what mouse capture looks like (on current `master`) when you do the following:
1. Press a (non-left) mouse button down with the cursor in the client area of the window and don't release it.
2. Move the cursor to the title bar.
3. Initiate a drag by pressing the left mouse button down.
4. Release the left moues button.
5. Move the mouse cursor outside the window.
6. Release the other mouse button.

That's (probably) the behaviour you want to match.

Reply:
You should also make sure nothing strange happens if `Window::set_drag_window` is called without any mouse buttons being pressed.

Reply:
Thank you, that helped a lot, everything seems to be in order except one big thing:
indeed this implementation causes a deadlock, it actually always happens, as soon as one starts to drag the window, it stops all incoming events, until the mouse button is released again.

I tried to figure it out but failed, any help would be appreciated.

Reply:
It's not really a deadlock. You're blocking the thread with `SendMessageW`, which will block until the message has been processed. Using `PostMessageW` instead seems to work fine.

One interesting "feature" I discovered is that you can initiate the drag when the cursor is outside the window by doing the following:
1. Press a non-left mouse button down inside the window.
2. Move the cursor outside the window.
3. Press the left mouse button.

I'm not sure if this is desirable.

Reply:
In the second case, you have try dragging twice. Once inside the window, and once outside.

Reply:
Alright, I reproduced it. Should we document this in your opinion?

Reply:
This outside-the-window-initiated dragging isn't something I see much sense in keeping around. I'd prefer to have a check for whether the mouse is inside the bounds of the window or not, since that's what I'd assume is expected on other platforms.

Reply:
So your comment prompted me to check how other platforms handle this, apparently this is pretty consistent except for Wayland.
This outside-the-window-initiated dragging makes it possible to drag a window from another window, so I think it's best to keep it like that and note that this doesn't work on Wayland, I will also add an example for this.